### PR TITLE
not output background object

### PR DESCRIPTION
**Pull request by YiwenShaoStephen**
Now only object class > 0 (not background) will be output. It solves problems that not all background pixels are merged into one object. This gives me a improvement from 0.37 to 0.40.

Also rewrite the visualization code so that only objects (not background) will have a mask and be totally transparent on other places.

### CODE REVIEW

File: scripts/waldo/data_visualization.py [hunk 1]
```diff
@@ -22,53 +23,34 @@ def visualize_mask(x, c, transparency=0.7, show_labels=True):
...
-    plt.imshow(color_mask, alpha=transparency)
-    if show_labels:
-        for objid in obj2center:
-            center = obj2center[objid]
-            color = colormap(shuffled_object_ids[objid])
-            plt.text(center[1] - 2, center[0] + 2, '{}'.format(objid), fontsize=7,
+    for i in range(1, mask.max() + 1):
+        b_mask = (mask == i)
+        base_img = np.ones((b_mask.shape[0], b_mask.shape[1], 3))
+        color = np.random.random((1, 3)).tolist()[0]
```
Comment (hhadian) [inline scripts/waldo/data_visualization.py]: Does this method make as distinguishable colors as a colormap does?

File: egs/dsb2018/v1/local/segment.py [hunk 2]
```diff
@@ -37,6 +37,12 @@
 parser.add_argument('--same-different-bias', type=float, default=0.0,
                     help='Bias for same/different probs in the segmentation '
                     'algorithm.')
+parser.add_argument('--merge-logprob-bias', type=float, default=0.0,
+                    help='A bias that is added to merge logprobs in the '
+                    'segmentation algorithm.')
+parser.add_argument('--prune-threshold', type=float, default=0.0,
+                    help='Threshold used in the pruning step of the '
+                    'segmentation algorithm. Higher values --> more pruning.')
```
Comment (danpovey) [inline egs/dsb2018/v1/local/segment.py]: I'm a little confused about these changes to segment.py and how they relate to other things we've been discussing.
Are you actually using these options?
And what is the pruning step?  I thought Hossein added something like this?

Comment (danpovey) [reply egs/dsb2018/v1/local/segment.py]: Oh I see, you are just passing in values that were already defined internally by Hossein.

File: scripts/waldo/data_visualization.py (hunk 1 above)
Comment (YiwenShaoStephen) [reply scripts/waldo/data_visualization.py]: Yes. Chances are really smaller that a random (1,3) float array is the same as a previous one.

**Comment by danpovey** [comment]
OK, merging.

**Pull request merged**